scylla-proxy/src/errors.rs

```diff
@@ -0,0 +1,62 @@
+use std::net::SocketAddr;
+
+use scylla_cql::frame::frame_errors::{FrameError, ParseError};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum DoorkeeperError {
```

Comment:
Could you write a brief example how I can use anyhow errors?

Reply:
Basically, you could get rid of enums in this file. Instead of using concrete error types, you use `anyhow::Error` or [`anyhow::Result<T>`](https://docs.rs/anyhow/latest/anyhow/type.Result.html) for functions that fail.

Returning errors with messages look like this:

```rust
return anyhow::anyhow!("Listen on socked {:?} failed", socket_addr);
```

Any error type converts to `anyhow::Error`. and you can attach additional context using the [`Context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html) trait:

```rust
let thing = function_that_can_fail().context("The function failed")?;
```

Refer to the documentation or `anyhow` for more details: https://docs.rs/anyhow/latest/anyhow/type.Result.html

Reply:
In some tests, I do match on errors. See `retries.rs:256` - DriverDisconnected is treated specially, as not a real error:
```
match res {
        Ok(()) => (),
        Err(ProxyError::Worker(WorkerError::DriverDisconnected(_))) => (),
        Err(err) => panic!("{}", err),
    }
```

As `thiserror`-based impl is ready and working, I would go for leaving it as is.